imagersite/imager_profile/handlers.py

```diff
@@ -0,0 +1,54 @@
+from .models import ImagerProfile
+from django.contrib.auth.models import User
+from django.dispatch import receiver
+from django.db.models.signals import post_save, post_delete
+
+
+@receiver(post_save, sender=User)
+def create_user_profile(sender, **kwargs):
+    """A User should always have an ImagerProfile"""
+    instance = kwargs.get('instance')
+    if not instance or kwargs.get('raw', False):
+        return
+    try:
+        instance.profile
+    except ImagerProfile.DoesNotExist:
+        instance.profile = ImagerProfile()
+        instance.profile.save()
+
+
+@receiver(post_save, sender=ImagerProfile)
+def create_user(sender, **kwargs):
+    """An ImagerProfile should always have a User"""
+    instance = kwargs.get('instance')
+    if not instance or kwargs.get('raw', False):
+        return
+    try:
+        instance.user
+    except User.DoesNotExist:
+        instance.user = User()
+        instance.user.save()
+
+
+@receiver(post_delete, sender=ImagerProfile)
+def rm_user_profile(sender, **kwargs):
+    """If an ImagerProfile is deleted, delete it's User too"""
+    instance = kwargs.get('instance')
+    if not instance:
+        return
+    try:
+        instance.user.delete()
+    except:
```

Comment:
Watch out for using a bare except clause.  You should decide (or learn) what errors might arise from calling `instance.user.delete` and catch them, but allow unexpected or unhandled errors to bubble up.  If you don't, something very bad could be happening in this code and you'd never learn of it directly.